src/bin/main.rs

```diff
@@ -61,6 +63,16 @@ struct Opts {
 const MAIN_LOOP_DEFAULT_SLEEP: u64 = 10;
 
 fn main() -> Result<()> {
+    // Guard against running as root. This check can be overridden by setting the PARSEC_ALLOW_ROOT
+    // environment variable.
+    if get_current_uid() == 0 && env::var("PARSEC_ALLOW_ROOT").is_err() {
```

Comment:
This can probably be added to the config file instead of an env var, and do the check after the file is parsed